src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+One thing to note here is that these tiers don't necessarily equate to object, file, or naming structure of the tokens, but is more conceptual based on *reference* structure of the tokens. The lowest level of tokens do not reference other tokens. They also do not have any contextual or semantic meaning, they are just *options*. 
+
+One important concept in Style Dictionary is that you can organize your token files however you want. Style Dictionary will search for all the token files based on the configuration you give it, and it will perform a deep merge of the token objects it finds in those files. This is the difference of *object* structure and *file* structure. File structure is purely for the authoring experience. What really matters is the *object* structure. 
+
+This structure is important because you don't want to have to replace every color token with a dark mode equivalent. Instead, only changing a thin semantic layer which component tokens reference. Let's take a look at a very small example:
+
+```json
+// core/brand tier
+{
+  "color": {
+    "core": {
+      "neutral": {
+        "0": { "value": "#ffffff" },
+        // ...
+      }
+    }
+  }
+}
+```
+
+```json
+// semantic/application-wide tier
+{
+  "color": {
+    "background": {
+      "primary": { "value": "{color.core.neutral.0.value}" }
+    }
+  }
+}
+```
+
+```json
+// component tier
+{
+  "component": {
+    "input": {
+      "background-color": { "value": "{color.background.primary.value}" }
+    }
+  }
+}
+```
+
+In this example we have 3 tokens: `component.input.background-color` references `color.background.primary` which references `color.core.neutral.0`. You could imagine many component tokens referencing `color.background.primary`. If we wanted to support dark mode all we would have to do is change the value of `color.background.primary` to reference a dark color like `color.core.neutral.1000`. This way we wouldn't need to create dark mode tokens for every component and instead update a thin semantic layer. 
+
+### Use descriptive names
+
+Another tip is to avoid specific color or tint in your tokens names for the middle tier. For example avoid `color.font.lighter` or `color.font.orange` because these token names are too tightly coupled with their values. `color.font.darker` is assumed to be *darker* than `color.font.dark`, but in dark mode this would probably not mean what you want it to mean. Instead opt for ordinal naming like `color.font.primary` and `color.font.secondary`. In light mode `secondary` would be lighter, but in dark mode it would be darker, both meaning less visual weight. Similarly, avoiding referencing the canonical color name like `orange` and instead using names that have some semantic meaning like `danger` or `warning`. 
```

Comment:
Similarly, avoiding referencing the canonical color name like `orange` and instead using names that have some semantic meaning like `danger` or `warning`.  > Similarly, avoid referencing the canonical color name like `orange` instead of using semantic meaning like `danger` or `warning`.